sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -209,17 +193,35 @@ public static ProxyOptions fromConfiguration(Configuration configuration, boolea
             ? Configuration.getGlobalConfiguration()
             : configuration;
 
-        for (BiFunction<Configuration, Boolean, ProxyOptions> loader : ENVIRONMENT_LOAD_ORDER) {
-            ProxyOptions proxyOptions = loader.apply(proxyConfiguration, createUnresolved);
+        return attemptToLoadProxy(proxyConfiguration, createUnresolved);
+    }
+
+    private static ProxyOptions attemptToLoadProxy(Configuration configuration, boolean createUnresolved) {
+        ProxyOptions proxyOptions;
+
+        // Only use system proxies when the prerequisite property is 'true'.
+        if (Boolean.parseBoolean(configuration.get(JAVA_SYSTEM_PROXY_PREREQUISITE))) {
+            proxyOptions = attemptToLoadSystemProxy(configuration, createUnresolved,
+                Configuration.PROPERTY_HTTPS_PROXY);
+            if (proxyOptions != null) {
+                return proxyOptions;
+            }
+
+            proxyOptions = attemptToLoadSystemProxy(configuration, createUnresolved, Configuration.PROPERTY_HTTP_PROXY);
             if (proxyOptions != null) {
                 return proxyOptions;
             }
         }
 
-        return null;
+        proxyOptions = attemptToLoadJavaProxy(configuration, createUnresolved, HTTPS);
+        if (proxyOptions != null) {
+            return proxyOptions;
+        }
```

Comment:
It would be good to add a log here to specify where the proxy was loaded from. Since this should only happen once per client creation, we can log at info level.

Reply:
Added logging. I used the verbose log level as this felt more like a debugging scenario log